src/config/plugin.ts

```diff
@@ -56,6 +56,33 @@ function processCommandIds(files: string[]): string[] {
   })
 }
 
+function determineCommandDiscoveryOptions(
+  commandDiscovery: string | CommandDiscovery | undefined,
+): CommandDiscovery | undefined {
+  if (!commandDiscovery) return
+
+  if (typeof commandDiscovery === 'string') {
+    return {globPatterns: GLOB_PATTERNS, strategy: 'pattern', target: commandDiscovery}
+  }
+
+  if (commandDiscovery.strategy === 'explicit') {
+    if (!commandDiscovery.target) throw new CLIError('`oclif.commandDiscovery.target` is required.')
+    return commandDiscovery
+  }
+
+  if (commandDiscovery.strategy === 'pattern') {
+    if (!commandDiscovery.target) {
+      throw new CLIError('`oclif.commandDiscovery.target` is required.')
+    }
+
+    return commandDiscovery
+  }
+}
```

Comment:
why are these separate check/throws for the same condition/error?

Reply:
Just a vestige of a previous implementation - I'll clean it up